### DISCUSSION TITLE
Microsoft's (New?)Universal FetchXML - Updated 08/10/2022

### DISCUSSION THREAD

**Discussion by WesleyTheMan** [rappen/FetchXMLBuilder#765] [Ideas]
[Microsoft's (New?) Universal FetchXML - Updated 08/10/2022](https://docs.microsoft.com/en-us/dynamics365/common-scheduler/developer/universal-fetchxml)

@rappen You're the GOAT - I'm a Noob here - does this change the game though? Seems significant.

Thank you!

From Link:
""""
UFX is an advanced query language that allows you to query data using dynamic FetchXML, shape, and prepare the resulting data for consumption by the Universal Resource Scheduling (URS) solution. This query language enables you to create custom queries to customize and extend the schedule board and schedule assistant filters to meet the unique business needs of the organization.

The resulting bag in XML:

XML

Copy
```
<bag>
    <first_account_name ufx-type="string">Fourth Coffee (sample)</first_acount_name>
</bag>
```
Certainly the most powerful aspect of a UFX Query is its ability to dynamically generate FetchXML based on input data.

In the sample below, we search for accounts by a value supplied by the user and available as a UFX Bag through the XPath $input variable. Notice the UFX if and value directives on the condition element.

XML

Copy
```
<bag xmlns:ufx="https://schemas.microsoft.com/dynamics/2017/universalfetchxml">
    <accounts ufx:source="fetch">
        <fetch top="10">
            <entity name="account">
                <filter>
                    <condition attribute="name" operator="like" ufx:if="$input/NameFilter">
                        <ufx:value select="$input/NameFilter" attribute="value" />
                    </condition>
                </filter>
            </entity>
        </fetch>
    </accounts>
</bag>
```
If the NameFilter property in the input bag contained %city% the produced FetchXML condition executed by Dynamics 365 would look like this.

XML

Copy
```
<condition attribute="name" operator="like" value="%city%" />
```
""""

**Comment by rappen** [comment]
Thanks @WesleyTheMan, and sorry about being slow...

I think it is like comparing apples and oranges... or maybe comparing grass with a cow.
A `bag` can have `FetchXml` in it, but `bag` is only for URS...
Makes sense?
